src/sniff.rs

```diff
@@ -0,0 +1,114 @@
+use crate::errors::*;
+use std::ffi::CString;
+use std::ffi::CStr;
+
+pub struct Cap {
+    handle: *mut pcap_sys::pcap,
+}
+
+pub struct Config {
+    pub promisc: bool,
+    pub immediate_mode: bool,
+}
+
+pub fn open(dev: &str, config: &Config) -> Result<Cap> {
+    let mut errbuf = [0i8; 256];
```

Comment:
Do not hardcode buffer size. Use this instead:
```rust
    assert!(std::mem::size_of::<usize>() >= std::mem::size_of::<u32>());
    // The above assert makes sure the `pcap_sys::PCAP_ERRBUF_SIZE as usize` below will not truncate.
    // It is required because `.try_into()` is not a const fn and `usize` does not implement `From<u32>`
    let mut errbuf = [0i8; pcap_sys::PCAP_ERRBUF_SIZE as usize];
```
`std::mem::size_of` is `const fn` so the assert will be eliminated by the optimizer.

Since this is repeated several times, it might make sense to use `static_assertions` crate to split this into its own `const fn`, but in this case it's probably overkill.

Reply:
Thanks, I've refactored this and included your snippet, can you double-check the new code doesn't introduce any additional issues?

Reply:
`func(errbuf.as_mut_ptr() as _)` looks like a potential footgun to me - or at least, something that I cannot immediately evaluate the impact of. Given that usize set to anything below `u32` while running an OS capable of libpcap is already super weird, and that the buffer size seems to have been hardcoded to 256 for ages that would fit even in a u16, I'd rather drop the extra complexity and let it truncate. This is so monumentally unlikely to go wrong that it's not worth even a bit of complexity.

Reply:
Ok, updated it again